source/ui/beamline/AMDeadTimeButton.cpp

```diff
@@ -142,3 +187,48 @@ bool AMDeadTimeButton::hasICRDataSource() const
 {
 	return !(inputCountSource_ == 0);
 }
+
+bool AMDeadTimeButton::hasAcquireTime() const
+{
+	return !(acquireTimeControl_ == 0);
+}
+
+double AMDeadTimeButton::getPercent() const
+{
+	double result = 0;
+
+	if (canDisplayPercentage())
+		result = 100*(1 - double(outputCountSource_->value(AMnDIndex()))/double(inputCountSource_->value(AMnDIndex())));
+
+	return result;
+}
+
+double AMDeadTimeButton::getCounts() const
+{
+	double result = 0;
+
+	if (canDisplayCounts())
+		result = double(inputCountSource_->value(AMnDIndex()));
+
+	return result;
+}
+
+double AMDeadTimeButton::getCountRate() const
+{
+	double result = 0;
+
+	if (canDisplayCountRate())
+		result = double(inputCountSource_->value(AMnDIndex())) / acquireTimeControl_->value();
```

Comment:
`canDisplayCountRate()` checks for whether or not the acquireTimeControl_ is 0, but this check could definitely be clearer.